dts: bindings: Move the dts clips to the examples

rruuaanng opened the pull request:
```
description: |
  SparkFun SerLCD Dot Character VFD Controller/Driver IC
  Example:
    &i2c1 {
      serlcd@72 {
        compatible = "sparkfun,serlcd";
        reg = <0x72>;
        columns = <16>;
        rows = <2>;
        command-delay-ms = <10>;
        special-command-delay-ms = <50>;
      };
    };
```
move to
```
examples:
  - |
    &i2c1 {
      serlcd@72 {
        compatible = "sparkfun,serlcd";
        reg = <0x72>;
        columns = <16>;
        rows = <2>;
        command-delay-ms = <10>;
        special-command-delay-ms = <50>;
      };
    };
```

rruuaanng committed:
dts: bindings: auxdisplay: Move the dts clips to the examples

Move the dts sample nodes from the binding `description`
into the `examples` block.

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng committed:
dts: bindings: charger: Move the dts clips to the examples

Move the dts sample nodes from the binding `description`
into the `examples` block.

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng commented:
~Let's wait for it to be merged.~ I almost forgot, I can address it in this PR.

rruuaanng committed:
edtlib: Fix the accidental merging of examples in the binding

Fix the example nodes in the examples block being accidentally
merged and overwritten during build.

Signed-off-by: James Roy <rruuaanng@outlook.com>

josuah reviewed (commented):
The text below the examples seems more like description to me in this case. What do you think?

Thank you for the change!

henrikbrixandersen reviewed (changes requested):
Today, the examples are rendered in the documentation (see e.g. https://docs.zephyrproject.org/latest/build/dts/api/bindings/can/nxp%2Cflexcan.html#dtbinding-nxp-flexcan).

With this change, the examples are no longer rendered in the documentation?

rruuaanng committed:
dts: bindings: cache: Move the dts clips to the examples

Move the dts sample nodes from the binding `description`
into the `examples` block.

Signed-off-by: James Roy <rruuaanng@outlook.com>

rruuaanng committed:
dts: bindings: timer: Move the dts clips to the examples

Move the dts sample nodes from the binding `description`
into the `examples` block.

Signed-off-by: James Roy <rruuaanng@outlook.com>

henrikbrixandersen commented:
> > Today, the examples are rendered in the documentation (see e.g. https://docs.zephyrproject.org/latest/build/dts/api/bindings/can/nxp%2Cflexcan.html#dtbinding-nxp-flexcan).
> > With this change, the examples are no longer rendered in the documentation?
> 
> Yes, I believe we should keep the `Description` focused on explaining the binding, and provide an `Example` in the documentation to show a sample node. I think it should go at the very bottom of the documentation page.

My point is, we can't merge this until the `examples` section is rendered in the documentation as well - otherwise this information, which is now visible to the user, will become hidden.

rruuaanng commented on dts/bindings/auxdisplay/gpio-7-segment.yaml:
```diff
@@ -86,3 +62,27 @@ properties:
       be selected.
+
+examples:
+  - |
```
https://www.asdf-format.org/projects/asdf-standard/en/1.0.2/schemas/yaml_schema.html, In `#style`

It's just a common YAML convention for multi-line text. Just like using `description`. :)

kartben commented:
@henrikbrixandersen please revisit now that https://github.com/zephyrproject-rtos/zephyr/commit/4c8cb73198809e25e48d599ddc405fbf15e89869 has been merged

kartben commented:
adding myself as an assignee as this is documentation as much as it is DT and DT maintainer is currently AWOL.

kartben commented on dts/bindings/can/nxp,flexcan-fd.yaml:
```diff
@@ -6,7 +6,13 @@ description: |
+
+include: ["nxp,flexcan.yaml", "can-fd-controller.yaml",
+          "pinctrl-device.yaml", "nxp,rdc-policy.yaml"]
+
```
@rruuaanng this needs fixing / likely shouldn't be there in the first place :)

henrikbrixandersen replied:
Correct. The includes should be there.

henrikbrixandersen reviewed (changes requested):
Please address the comments regarding the unrelated changes to the NXP FlexCAN FD binding.

rruuaanng committed:
dts: bindings: can: Move the dts clips to the examples

Move the dts sample nodes from the binding `description`
into the `examples` block.

Signed-off-by: James Roy <rruuaanng@outlook.com>

thedjnK reviewed (approved):
auxdisplay parts OK
